src/diffusers/models/vae.py

```diff
@@ -429,23 +484,35 @@ def __init__(
         self.quant_conv = torch.nn.Conv2d(2 * latent_channels, 2 * latent_channels, 1)
         self.post_quant_conv = torch.nn.Conv2d(latent_channels, latent_channels, 1)
 
-    def encode(self, x):
+    def encode(self, x, return_dict: bool = True):
         h = self.encoder(x)
         moments = self.quant_conv(h)
         posterior = DiagonalGaussianDistribution(moments)
-        return posterior
 
-    def decode(self, z):
+        if not return_dict:
+            return (posterior,)
+
+        return AutoEncoderKLOutput(latent_dist=posterior)
```

Comment:
```suggestion
        return AutoencoderKLOutput(latent_dist=posterior)
```

Renamed as per a previous comment.